cranelift/assembler-x64/meta/src/lib.rs

```diff
@@ -11,42 +11,17 @@ use std::process::Command;
 
 /// Generate the assembler `file` containing the core assembler logic; each of
 /// the DSL-defined instructions is emitted into a Rust `enum Inst`.
-pub fn generate_rust_assembler<P: AsRef<Path>>(dir: P, file: &str) -> PathBuf {
-    generate(dir, file, generate::rust_assembler, Language::Rust)
-}
-
-/// Generate a macro containing builder functions for the assembler's ISLE
-/// constructors; this uses the `build` module emitted in
-/// [`generate_rust_assembler`].
-pub fn generate_isle_macro<P: AsRef<Path>>(dir: P, file: &str) -> PathBuf {
-    generate(dir, file, generate::isle_macro, Language::Rust)
-}
-
-/// Generate the ISLE definitions; this provides ISLE glue to access the builder
-/// functions from [`generate_isle_macro`].
-pub fn generate_isle_definitions<P: AsRef<Path>>(dir: P, file: &str) -> PathBuf {
-    generate(dir, file, generate::isle_definitions, Language::Isle)
-}
-
-/// Helper for emitting generated lines into a formatted file.
 ///
 /// # Panics
 ///
 /// This function panics if we cannot update the file.
-fn generate<P: AsRef<Path>>(
-    dir: P,
-    file: &str,
-    generator: fn(&mut Formatter, &[dsl::Inst]),
-    lang: Language,
-) -> PathBuf {
+pub fn generate_rust_assembler<P: AsRef<Path>>(dir: P, file: &str) -> PathBuf {
     let out = dir.as_ref().join(file);
     eprintln!("Generating {}", out.display());
-    let mut fmt = Formatter::new(lang);
-    generator(&mut fmt, &instructions::list());
+    let mut fmt = Formatter::new(Language::Rust);
+    generate::rust_assembler(&mut fmt, &instructions::list());
     fmt.write(file, dir.as_ref()).unwrap();
-    if matches!(lang, Language::Rust) {
-        rustfmt(&out);
-    }
+    rustfmt(&out);
```

Comment:
FWIW, we removed the invocation of `rustfmt` in #9901 for the main meta crate; it's an artifact of an earlier time when we had the code checked into git, but we don't now, and it can cause issues in certain build environments -- probably ok to remove it here too?